chains/evm/calls/bridge.go

```diff
@@ -120,43 +119,43 @@ func ParseIsRelayerOutput(output []byte) (bool, error) {
 	return *b, nil
 }
 
-func Deposit(client ChainClient, fabric TxFabric, bridgeAddress, recipient common.Address, amount *big.Int, resourceID types.ResourceID, destDomainID uint8) error {
+func Deposit(client ClientDispatcher, fabric TxFabric, gasPriceClient GasPricer, bridgeAddress, recipient common.Address, amount *big.Int, resourceID types.ResourceID, destDomainID uint8) error {
```

Comment:
It would kinda make sense to me instead of passing the client, fabric and gasPriceClient in every function to just have
a `Transactor` interface that just has `Transact` function and configuring the client, fabric and gasPrice on constructor of Transactor.